### DISCUSSION TITLE
brave: the entire user home is accessible (dbus)

### DISCUSSION THREAD

**Discussion by firejailaddssecuirty** [netblue30/firejail#4795] [Q&A] [answered]
```
$ firejail --list
8587:fedora::/usr/bin/firejail /usr/bin/brave-browser-stable
```
When I test the sandbox by using CTRL+O (open) the entire /home/username is accessible. This is not the case with Firefox. Why is the firejail profile for Brave so relaxed ?

`Brave Browser 96.1.33.106 `
`
firejail version 0.9.66`

`Fedora 35`

```
$ cat /etc/firejail/brave-browser-stable.profile 
# Firejail profile alias for brave (release channel)
# This file is overwritten after every install/update
# Persistent local customizations
include brave-browser-stable.local
# Persistent global definitions
# added by included profile
#include globals.local

# Redirect
include brave.profile
```

**Comment by rusty-snake** [comment, marked as answer]
The file-open dialog may use portals, can you enter `file:///home/username` in the urlbar.

**Reply by firejailaddssecuirty** [reply]
![brave](https://user-images.githubusercontent.com/93089434/147331033-6076f387-a5c7-4d40-a7b1-56a87ae2debb.png)

**Reply by firejailaddssecuirty** [reply]
@rusty-snake 
Sorry for wasting your time. I don't understand when I use CTRL+O under Firefox nothing but only ~/Downloads can be read but under Brave the entire /home/username can be read but when file:///home/username is used in the urlbar I get confirmation that isolation is working properly.
What's the explanation for this ?

**Reply by rusty-snake** [reply]
xdg-desktop-portals (https://docs.flatpak.org/en/latest/desktop-integration.html#portals). The view you get with the file-open dialog isn't the view Brave has. Brave has only access to the files you see via `file:///`.

If you set `widget.use-xdg-desktop-portal=true` on `about:config` you get the same in firefox.
Example: `firejail --noprofile --shell=none --private --env=GTK_USE_PORTALS=1 /usr/bin/firefox --no-remote`
